android/src/main/java/com/ioreactnativecieid/IoReactNativeCieidModule.kt

```diff
@@ -28,9 +29,40 @@ class IoReactNativeCieidModule(reactContext: ReactApplicationContext) :
 
   override fun getName() = NAME
 
+  private fun isSignatureValid(packageName: String, signature: String): Boolean {
+    val pm = reactApplicationContext.packageManager
+    val packageInfo = if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.P) {
+      pm.getPackageInfo(packageName, PackageManager.GET_SIGNING_CERTIFICATES)
+    } else {
+      @Suppress("DEPRECATION")
+      pm.getPackageInfo(packageName, PackageManager.GET_SIGNATURES)
+    }
+
+    val signatures = if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.P) {
+      packageInfo.signingInfo.apkContentsSigners
+    } else {
+      @Suppress("DEPRECATION")
+      packageInfo.signatures
+    }
+
+    val sha256List = signatures.map {
+      val digest = MessageDigest.getInstance("SHA-256")
+      digest.update(it.toByteArray())
+      // Converts bytes in hexadecimal uppercase string
+      digest.digest().joinToString(":") { byte ->
+        "%02x".format(byte).uppercase()
+      }
+    }
+    // Returns the first sha256 signature
+    return sha256List.contains(signature)
+  }
```

Comment:
What about leveraging extension function for more readability and maintenance? What do you think? 

```suggestion
  private fun isSignatureValid(packageName: String, signature: String): Boolean {
    val pm = reactApplicationContext.packageManager
    val packageInfo = pm.getPackageInfoCompat(packageName)

    val signatures = packageInfo.getSignaturesCompat()

    val sha256List = signatures.map { it.toSha256() }

    // Check if the given signature is in the SHA-256 list
    return sha256List.contains(signature)
  }

  // Extension function to handle API compatibility for getPackageInfo
  private fun PackageManager.getPackageInfoCompat(packageName: String): PackageInfo {
    return if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.P) {
      getPackageInfo(packageName, PackageManager.GET_SIGNING_CERTIFICATES)
    } else {
      @Suppress("DEPRECATION")
      getPackageInfo(packageName, PackageManager.GET_SIGNATURES)
    }
  }

  // Extension function to handle API compatibility for getting signatures
  private fun PackageInfo.getSignaturesCompat(): Array<Signature> {
    return if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.P) {
      signingInfo.apkContentsSigners
    } else {
      @Suppress("DEPRECATION")
      signatures
    }
  }

  // Extension function to convert a signature to its SHA-256 hash
  private fun Signature.toSha256(): String {
    val digest = MessageDigest.getInstance("SHA-256")
    return digest.digest(toByteArray())
      .joinToString(":") { byte -> "%02x".format(byte).uppercase() }
  }
```

Reply:
I can only agree with you; thank you for the suggestion, I wasn't aware of this approach!